### PR TITLE
Fix for empty label on expand button

### LINKED ISSUE #128: Menu down arrow is moved to the right when menu loaded without label
screen shot 2018-04-04 at 10 18 59 pm

### PR DESCRIPTION
**Pull request by seangates**
## Description
When the label is empty (or missing altogether) it will apply the correct margins to the icon.

## References
Closes #128 

## Screenshots
image

### CODE REVIEW

**Comment by ianmcburnie** [comment]
Can we save this into 4.1.0? v4.0.0 is frozen in terms of QA and regression testing cycles.

File: src/less/button/ds6/button-base.less [hunk 1]
```diff
@@ -308,4 +308,12 @@ span.expand-btn__icon {
     background-size: contain;
     flex-shrink: 0;
     margin-left: 8px;
+
+    button.expand-btn span.expand-btn__cell > &:only-child {
```
Comment (ianmcburnie) [inline src/less/button/ds6/button-base.less]: Selector can be reduced from `button.expand-btn span.expand-btn__cell` to just `span.expand-btn__cell`... unless you need the specificty.

Comment (senthilp) [reply src/less/button/ds6/button-base.less]: Yes, they should be reduced. That is one of the advantages of using BEM syntax. `span.expand-btn__cell` will end up being a unique selector.

**Comment by seangates** [comment]
Well ... I was thinking of leaving this undocumented, to be honest. It's not a pattern I've seen from the platform design group.

Thoughts?

**Comment by ianmcburnie** [comment]
Hmm, well we have to have it somewhere to know if we cause a regression or not. Maybe append it to the expand button section of button module?

File: docs/_includes/ds6/button.html [hunk 2]
```diff
@@ -210,12 +210,26 @@ <h3 id="button-expand">Expand Button</h3>
                 </span>
             </button>
         </div>
+        <div class="demo__inner">
+            <button aria-expanded="false" class="expand-btn expand-btn-example">
+                <span class="expand-btn__cell">
+                    <span></span>
```
Comment (ianmcburnie) [inline docs/_includes/ds6/button.html]: Does this empty span need to stay there?

Comment (seangates) [reply docs/_includes/ds6/button.html]: It's not required, no.

**Comment by seangates** [comment]
@ianmcburnie Fixed. Please review once more.

File: docs/_includes/ds6/button.html [hunk 3]
```diff
@@ -210,12 +210,25 @@ <h3 id="button-expand">Expand Button</h3>
                 </span>
             </button>
         </div>
+        <div class="demo__inner">
+            <button aria-expanded="false" class="expand-btn expand-btn-example" aria-label="Expand/Collapse">
+                <span class="expand-btn__cell">
+                    <span></span>
```
Comment (ianmcburnie) [inline docs/_includes/ds6/button.html]: I still see an empty span here. Remove that and we should be good to merge.

File: docs/_includes/ds6/button.html [hunk 4]
```diff
@@ -210,12 +210,25 @@ <h3 id="button-expand">Expand Button</h3>
...
+        </div>
         {% highlight html %}
 <button aria-expanded="false" class="expand-btn">
     <span class="expand-btn__cell">
         <span>Button</span>
         <span class="expand-btn__icon"></span>
     </span>
+</button>
+<button aria-expanded="false" class="expand-btn" aria-label="Expand/Collapse">
+    <span class="expand-btn__cell">
```
Comment (ianmcburnie) [inline docs/_includes/ds6/button.html]: Quick question though: I wonder whether we even need the cell container when we just have an icon. If I remember correctly, the cell is there to align text with an icon, so possibly not needed for an icon only case. I could be wrong though. Potentially we should be able to put any icon into a button and it look okay.

**Comment by seangates** [comment]
@ianmcburnie I added a second example without the cell.

**Comment by ianmcburnie** [comment]
I think we can drop support for the first example with the cell now then. Cell-less seems the way to go.

**Comment by seangates** [comment]
@ianmcburnie Okay, removed the cell version. Now only the cell-less version remains.

**Comment by ianmcburnie** [comment]
As discussed offline, I think we can reduce to:

```css
&:only-child {
        display: flex;
        margin: 0;
    }
```

Even though this will cover one less edge-case (icon span preceded by empty span)

**Comment by seangates** [comment]
@ianmcburnie Please review when you get a moment. I think we're ready to go, though. 👍

**Pull request merged**